### PR TITLE
Add helpers to utils

### PR DESCRIPTION
**Pull request by lkwdwrd**
Adds some server utils to start moving some boiler plate over here.

### CODE REVIEW

File: database/db.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+// Migrate runs all migrations found in migrationDir against db.
+func (db *DB) Migrate() error {
+	err := migrations.Run(db.Client, "", []string{"", "migrate"})
+	return err
+}
+
+// dbLogger implements the DBLogger pattern for the go-pg module
+// https://github.com/go-pg/pg/wiki/FAQ#how-can-i-viewlog-queries-this-library-generates
```
Comment (willmichael) [inline database/db.go:41]: cool addition

File: server/server.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// Middleware is a function that decorates an http.Handler
+type Middleware func(handler http.Handler) http.Handler
```
Comment (willmichael) [inline server/server.go:10]: Middleware type living in middleware file makes a little more sense to me

File: server/server.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// ApplyMiddleware creates a final http Handler with a slice of middleware functions applied
+func ApplyMiddleware(handlers []Middleware, final http.Handler) http.Handler {
+	for _, handler := range handlers {
+		final = handler(final)
+	}
+	return final
+}
+
+// ServeMux serves a set of endpoints defined in a ServerMux with the defined middleware applied
+func ServeMux(httpd *http.ServeMux, mw []Middleware, servicePort string) *http.Server {
```
Comment (willmichael) [inline server/server.go:21]: Curious, is there a reason we're implementing this ourselves instead of using something like gorilla mux?

**Review by willmichael — APPROVED** [review]

File: database/db.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+// New returns a new DB object which wraps
+// a connection to the database specified in config
+func New(config DBConfig) DB {
+	db := pg.Connect(&pg.Options{
+		Addr:     config.Address,
+		User:     config.User,
+		Database: config.Database,
+		Password: config.Password,
+	})
+	if os.Getenv("QUERY_DEBUG") == "ON" {
```
Comment (galxy25) [inline database/db.go:69]: This should come from a field in config so its explicitly set and not accidentally overridden/clobbered.

File: server/server.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	// Create an http server with the provided config
+	server := &http.Server{
+		Addr: fmt.Sprintf(":%s", servicePort), //host:port for the server to listen on, defaults to localhost
+		// Log all requests made to this server
```
Comment (galxy25) [inline server/server.go:25]: comments on lines 25-27 should be removed

File: server/server.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// Middleware is a function that decorates an http.Handler
+type Middleware func(handler http.Handler) http.Handler
+
+// ApplyMiddleware creates a final http Handler with a slice of middleware functions applied
+func ApplyMiddleware(handlers []Middleware, final http.Handler) http.Handler {
```
Comment (galxy25) [inline server/server.go:13]: I feel this function and the type Middleware belong in middleware.go

File: database/db.go [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+type DB struct {
+	Client *pg.DB
+}
+
+// Close closes a connection to a database. Once close has been called calling other methods on db will error.
+func (db *DB) Close() {
+	db.Client.Close()
+}
+
+// Migrate runs all migrations found in migrationDir against db.
```
Comment (galxy25) [inline database/db.go:34]: Can you document the additional requirements needed to make this work? Thinking specifically in regards to: https://github.com/tozny/client-service/pull/1/files#diff-65d1a23063053a9c8ba9175c4e25dcda

**Review by galxy25 — APPROVED** [review]

**Pull request merged**